packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -930,6 +930,11 @@ export const ComboBoxMixin = (subclass) =>
           this.opened || this.autoOpenDisabled
             ? this.$.dropdown.indexOfLabel(this.filter)
             : this._indexOfValue(this.value, this.filteredItems);
+
+        // see https://github.com/vaadin/web-components/issues/2615
+        if (this.value && this._focusedIndex >= 0 && this.filteredItems[this._focusedIndex] === this.value) {
```

Comment:
`this.filteredItems[this._focusedIndex] === this.value` looks a bit suspicious since `this.filteredItems` is of type `Array<TItem>` whereas `this.value` is of type `string`. Should probably get the item value using `_getItemValue(item)` and then using the resulting `string` for the comparison.

Reply:
Like this?
```suggestion
        if (this.value && this._focusedIndex >= 0 && this._getItemValue(this.filteredItems[this._focusedIndex]) === this.value) {
```

Reply:
Added a change with a nested `if` since it was getting a bit too long.